Marlin/Marlin_main.cpp

```diff
@@ -5832,10 +5832,17 @@ inline void gcode_M17() {
   ) {
     if (move_away_flag) return false; // already paused
 
-    if (!DEBUGGING(DRYRUN) && thermalManager.tooColdToExtrude(active_extruder) && unload_length > 0) {
-      SERIAL_ERROR_START;
-      SERIAL_ERRORLNPGM(MSG_TOO_COLD_FOR_M600);
-      return false;
+    if (!DEBUGGING(DRYRUN) && unload_length > 0) {
+      #if ENABLED(PREVENT_COLD_EXTRUSION)
+        if (!thermalManager.allow_cold_extrude && 
+            thermalManager.degTargetHotend(active_extruder) < thermalManager.extrude_min_temp) {
+          SERIAL_ERROR_START;
+          SERIAL_ERRORLNPGM(MSG_TOO_COLD_FOR_M600);
+          return false;
+        }
+      #endif
```

Comment:
The previous implementation of `M600` would helpfully start heatup for you and go to the wait-for-heatup screen. Then it would do the same for you if the nozzle timed out and started to cool again. This seems like a step backward to insist that the target temperature already be set, especially as we still have to make sure the nozzle has reached the set temperature anyway and we still have to deal with reheating after a timeout.

So, let's try to simply restore everything that `M600` code used to do instead of taking this more timid route. A lot of good work was done by @petrzjunior and I hate to see it tossed out.